### PR TITLE
Fix running test files with periods in the name

### LINKED ISSUE #29502: ClassNotFoundException when having periods in the bal file when running test
**Description:**
I have the following test directory structure

└── tests
    └── file.tests.bal

when running tests following error observed.

error: java.lang.ClassNotFoundException: waruna.FileManager.0_1_0.tests.file.tests

### PR DESCRIPTION
**Pull request by Ibaqu**
## Purpose
Fixes ClassNotFoundException when running files

Fixes #29502

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by Ibaqu** [comment]
With the following fix, can now test files with periods in them. 

- Applies to both single bal files and projects
- Multiple periods in the file are also supported

File: cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/TestCommandTest.java [hunk 1]
```diff
@@ -72,6 +72,18 @@ public void testTestBalFile() throws IOException {
         testCommand.execute();
     }
 
+    @Test(description = "Test a valid ballerina file with periods in the file name")
+    public void testTestBalFileWithPeriods() throws IOException {
```
Comment (pramodya1994) [inline cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/TestCommandTest.java:76]: Cant we remove this `IOException`, seems like it's never thrown.

File: cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/TestCommandTest.java [hunk 2]
```diff
@@ -72,6 +72,18 @@ public void testTestBalFile() throws IOException {
...
+        Path validBalFilePath = this.testResources.resolve("valid-test-bal-file").resolve("sample.tests.bal");
+
+        System.setProperty("user.dir", this.testResources.resolve("valid-test-bal-file").toString());
```
Comment (pramodya1994) [inline cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/TestCommandTest.java:79]: Can use `ProjectConstants.USER_DIR`

**Review by pramodya1994 — CHANGES_REQUESTED** [review]
Added 2 minor suggetions

File: cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/TestCommandTest.java (hunk 1 above)
Comment (Ibaqu) [reply cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/TestCommandTest.java]: Removed `throws IOException`

File: cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/TestCommandTest.java (hunk 2 above)
Comment (Ibaqu) [reply cli/ballerina-cli/src/test/java/io/ballerina/cli/cmd/TestCommandTest.java]: Changed all instances of "user.dir" to ProjectConstants.USER_DIR

**Review by azinneera — APPROVED** [review]

**Review by pramodya1994 — APPROVED** [review]

**Pull request merged**